capstone/src/main/java/com/google/sps/servlets/utils/ServletUtil.java

```diff
@@ -26,4 +42,31 @@ public static String getNameByEmail(String email) {
     }
     return entity.getProperty(Constants.PROPERTY_NAME).toString();
   }
+
+  public static String getRedirectUri(HttpServletRequest req) {
+    // TODO: change redirect URI when web app is deployed.
+    // If you want to run this locally, you will need to replace this with your dev server URI
+    // - then add "/oauth2callback" to the end of it and add that to you API console under
+    // Authorized URIs.
+    return "https://8080-4417b0ad-e7ff-4d2f-acc8-c9ddcf7b56d9.us-west1.cloudshell.dev/oauth2callback";
+  }
+
+  public static GoogleAuthorizationCodeFlow newFlow() throws IOException {
+    return new GoogleAuthorizationCodeFlow.Builder(
+            HTTP_TRANSPORT,
+            JSON_FACTORY,
+            getClientCredential(),
+            Collections.singleton(CalendarScopes.CALENDAR))
+        .setDataStoreFactory(AppEngineDataStoreFactory.getDefaultInstance())
+        .setAccessType("offline")
```

Comment:
nit: Add constant global variable for "offline" ? (just in this file, not Constants.java)